src/features/common/Layout/UserPropsContext.tsx

```diff
@@ -120,9 +120,8 @@ export const UserPropsProvider: FC = ({ children }) => {
   }, [token, refetchUserData]);
 
   React.useEffect(() => {
```

Comment:
As `impersonationModeOn` is a dependency, and we are calling `setIsImpersonationModeOn` within the useEffect, unnecessary calls will be made to `setIsImpersonationModeOn` (at least 2 times while loading impersonation mode initially from local storage), and in the worst case we could end up in a loop.

Could we change this to:

React.useEffect(() => {
  const impersonationData = localStorage.getItem('impersonationData');
  if (impersonationData !== null && !isImpersonationModeOn) {
    setIsImpersonationModeOn(true);
  } else if (impersonationData === null && isImpersonationModeOn) {
    setIsImpersonationModeOn(false);
  }
}, []);

Would removing the dependency cause an issue?

Reply:
outdated - closing this